
---


## What is epigenomics?

.center[.large[What makes them different?]]

.center[.image-60[![](img/epi1.png)]]

<!---
<div class="centered">
<img src="img/epi1.png" style="max-height: 60%; max-width: 60%"/>
</div>
--->

Genes are **expressed** differently during different **stages** and in different **tissues**.

???

- Something that we want to understand is how that genetic variation leads to phenotypic variation.
- A major part of understanding that, comes from understanding how regulation works.

---
class: split-50, middle

## What is epigenomics?

.column[
DNA methylation is a chemical modification of DNA, involved in *gene expression* regulation.
]

.column[
.center[.image-30[![](img/epi4.png)]]
]

<!--
<div class="centered">
<img src="img/epi3.png" style="max-height: 60%; max-width: 60%"/>
<img src="img/epi4.png" style="max-height: 60%; max-width: 60%"/>
</div>
-->


<!--
<footer class="source">[Robertson and Wolffe, Nat Rev Genet, 2000]</footer>
-->

???

- DNA methylation plays a major role in this regulation process
- Plus, we know how it is inherited from cell to cell, cell-fate
- invovled in differentiation
- can give us an important idea of this process in development and disease
- plasticity

---

## Probing DNA methylation

.center[
.image-100[![](img/meth_pileup.png)]
]

---
class: split-50, middle

## Probing DNA methylation

.column[
.image-90[![](img/measure_meth.png)]
]

.column[
- local-likelihood smoothing method
- **high-frequency smoothing** estimates local methylation structure (small domains)
- **low-frequency smoothing** estimates long-range methylation structure (large domains)
]

<!--
<footer class="source">
  Nature Genetics, 2011\br
  Bioinformatics, 2013
</footer>
-->

---

## DNA methylation in cancer

.center[.image-100[![](img/block.png)]]

Large blocks of **hypo-methylation** in colon cancer

<!--
<footer class="source">Nat. Genetics, 2011</footer>
-->


- consistent but partially methylated in normal tissues
- levels are inconsistent in cancer, but boundaries are conserved across samples
- overlaps with other important genomic domains
- genes within these blocks are **tissue-specific**

---
class: middle
exclude: true

.center[.image-100[![](img/timp1.png)]]

**Hypo-methylation** blocks observed across five solid tumor types.

<!--
<footer class="source">Genome Medicine, 2014</footer>
-->

---
class: middle

.center[.image-80[![](img/antiprofile.png)]]


Genes with **hyper-variable** expression in colon cancer are enriched within these blocks.

<!--
<footer class="source">Nat. Genetics, 2011</footer>
-->

---
class: middle
exclude: true

.center[.image-100[![](img/timp2.png)]]

Gene expression **hyper-variability** enriched in hypo-methylation blocks in other cancer types.

<!--
<footer class="source">Genome Medicine, 2014</footer>
-->

---
class: middle
exclude: true

<div class="centered">
<img src="img/barcode.png" style="max-height: 60%; max-width: 60%"/>
<img src="img/barcode2.png" style="max-height: 35%; max-width: 35%"/>
</div>

Genes with **consistent hyper-variable expression** across tumors are _tissue-specific_.

<!--
<footer class="source">BMC Bioinformatics, 2013</footer>
-->

---

## Summary

- large domains of methylation loss are a stable mark across cancer types
- gene expression hyper-variability is enriched within these domains
- hyper-variable genes within these regions are tissue-specific and involved in cellular fate

