
---
class: split-50

## Software for statistical analysis

.column[
State-of-the-art computational and statistical analysis
platform  

We develop and apply methods for these analyses in this platform  

Our collaborators do analysis in this platform with us

.image-30[![](img/rbioc.png)]

]

.column[
.center[
`minfi`  
`bumphunter`  
`antiProfiles`  
`HTShape`  
`qsmooth`  
`metagenomeSeq`  
`metagenomeFeatures`  
`Rcplex`  
`Rcsdp`
]
]

---

### Exploratory and collaborative analysis

- *Data transformation and modeling*: data smoothing, region finding
(R/Bioconductor: `Bsmooth`, `minfi`)

--

- *Exploration*: search by gene, search by overlap

--

- *Contextual analysis*: overlap with other data (our own, other labs, UCSC, ensembl)

---

## Genomic Data Science  

- We have unprecedented ability to measure
- and lots of publicly available data to contextualize it

.center[
<img src="img/hadley.png" style="max-height: 50%; max-width: 50%"/>
<img src="img/hadley2.png" style="max-height: 30%; max-width: 30%"/>
]

.source[[H. Wickham]]

---

.center[.image-40[![](img/epiviz_logo_large_white.png)]]


### Integrative, visual and computational exploratory analysis of genomic data

- Browser-based
- Interactive
- Integration of data
- Reproducible dissemination
- Communication with R/Bioconductor: `epivizr` package

e.g.: http://epiviz.cbcb.umd.edu/?ws=YOsu0RmUc9l

.source[[Nat. Methods, 2014]]

---
exclude: true
class: center, middle

<iframe src="http://epiviz.cbcb.umd.edu/?ws=clHVLu0BQti" height="100%" width="100%" frameborder="0"></iframe>

---
class: middle

## Creativity in exploration

We are building software applications to support creative exploratory analysis of large genome-wide datasets...

.center[.image-60[![](img/terry.png)]]

.source[[T. Speed]]

---
class: middle, split-30

### Summarization

.column[
Summarize integrated measurements (computed on data subsets)
]

.column[
.center[
<video height="450px" controls>
  <source src="movies/transform_aggregate.mp4" type="video/mp4">
</video>
]
]

---
class: middle, split-40

### Statistically-guided exploration

.column[
Calculate a statistic of interest  
Interactively explore data based on statistic
]

.column[
```{r, tidy=FALSE, eval=FALSE}
# Get tumor methylation base-pair data
m <- assay(se)[,"tumor"]

# Compute regions with highest variability across cpgs
region_stat <- calcWindowStat(m, step=25, 
                              window=80, stat=rowSds)
s <- region_stat[,"stat"]

# get locations in decreasing order
o <- order(s, decreasing=TRUE)
indices <- region_stat[o, "indices"]
slideShowRegions <- rowRanges(se)[indices] + 1250000L
mgr$slideshow(slideShowRegions)
```
]

---
class: center, middle

<video height="550" controls>
  <source src="movies/guided_exploration_v2.mp4" type="video/mp4">
</video>

---
class: middle, split-60
exclude: true

### Dynamically extensible

.column[
Integrate new data types and add new visualizations.

- Based on classic "three-table" design in genomic data analysis
- **Data providers define coordinate space**
]

.column[
.center[
<img src="img/datatypes.png" style="max-width=50%; min-width=50%"/>
]
]

---
exclude: true
<div class="centered">
<video height="550" controls>
  <source src="movies/finance.mp4" type="video/mp4">
</video>
</div>

---
class: middle

### Visualization design goals

- Context 
  - Integrate and align multiple data sources; navigate; search
  - *Connect*: brushing
  - *Encode*: map visualization properties to data on the fly
  - *Reconfigure*: multiple views of the same data

.source[[Perer & Shneiderman]]

---
class: middle

### Visualization design goals

- Data
  - *Select and filter*: tight-knit integration with R/Bioconductor; 
  - (current work) filters on visualization propagate to data environment
- Model
  - New 'measurements' the result of modeling; perhaps suggested by data context

.source[[Perer & Shneiderman]]

---
class: middle

.center[
<img src="img/hadley.png" style="max-height: 50%; max-width: 50%"/>
<img src="img/hadley2.png" style="max-height: 30%; max-width: 30%"/>
]

.source[[H. Wickham]]

Specific challenge for epigenomic data analysis is how to integrate  *many relevant sources of contextual data*

- Easily access/integrate *contextual* data
- Driven by exploratory analysis of *immediate* data
- Iterative process
- Visual and computational exploration go hand in hand

