<!DOCTYPE html>
<html>
  <head>
    <title>Interactive visual and computational exploratory analysis of epigenetic data</title>
    <meta charset="utf-8">
    <meta name="author" content="Héctor Corrada Bravo" />
    <meta name="date" content="2017-06-14" />
    <link href="libs/remark-css-0.0.1/example.css" rel="stylesheet" />
    <link rel="stylesheet" href="custom.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">


class: title-slide, center, middle
count: false

.banner[![](img/epiviz.png)]

.title[Interactive, visual and statistical exploratory analysis of epigenomic data]

.author[Héctor Corrada Bravo]

.other-info[
Center for Bioinformatics and Computational Biology  
University of Maryland, College Park, USA  
VIZBI '17: June 14, 2017
]

.logo[![](img/logo.png)]

---
class: split-50

## What does my group do?

.column[
Study the **molecular** basis of *variation* in development and disease


Using **high-throughput** experimental methods  
]

.column[.image-80[![](img/stickmen.png)]]







---

## Lessons Learned

### R/Bioconductor connection is a real success

- Fully scriptable UI  
  - interactive viz weaved with statistical analysis
  - _Reproducible_? (_Reconstructable_, really)
  
- Integrate interactive visualization as part of literate
  programming with Rmarkdown
  
- Bioconductor community as a resource: **think about role of interactive visualization in day to day data analysis**
  
---
class: split-60, middle

## Survey the community

### Online Bioconductor community survey

.column[
- 65 respondents  
- Mostly expert level users and developers
- Data analyst rather than Biomedical scientist
]

.column[
.image-40[![](img/tweet.png)]
]

About half has used interactive visualization previously  
Predominantly through shiny: https://shiny.rstudio.com


---

### Issues faced creating interactive visualizations





&lt;img src="index_files/figure-html/plot_q26-1.png" width="1056" /&gt;

---

### Would like to see in genomic interactive visualization platform



&lt;table class=survey_table&gt;
 &lt;thead&gt;
  &lt;tr&gt;
   &lt;th style="text-align:left;"&gt; response &lt;/th&gt;
   &lt;th style="text-align:right;"&gt; num. respondents &lt;/th&gt;
  &lt;/tr&gt;
 &lt;/thead&gt;
&lt;tbody&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; The interactive visualization tool should be a very flexible and extendable system that I can use to develop and create customized workflows. &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 26 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; I would use the interactive tool as a standalone application that would quickly let me load, visualize and analyze data. As an end user, I am mostly interested in the usability of the application. &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 14 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; I don't know &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 5 &lt;/td&gt;
  &lt;/tr&gt;
&lt;/tbody&gt;
&lt;/table&gt;

---

## Takeaways

- Authoring, extensibility, infrastructure
- Collaboration within teams
- Interactive publishing and deposition

---
class: split-50

## Epiviz components (in development)

Drop in epiviz visualization components

.column[
- Connected to either webserver or local R environment
- Can be embedded into HTML
- Reusable components
]

.column[
```html
&lt;head&gt;
&lt;link rel="import" href="epiviz-charts"&gt;
&lt;/head&gt;

&lt;body&gt;
...
&lt;epiviz-navigation chr="chr6" ...&gt;
  &lt;epiviz-genes-track ...&gt;
  &lt;epiviz-scatter-plot x="cancer" y="normal"&gt;
&lt;/epiviz-navigation
&lt;/body&gt;
```
]

.footer[http://github.com/epiviz/epiviz-chart]

---

**Acknowledgements**  

Justin Wagner, Jayaram Kancherla (CBCB)    
Florin Chelaru (now at Google), Joseph Paulson (now at Genentech)  
Feinberg Lab &amp; K. Hansen (JHU), R. Irizarry (Harvard)

Funding: NIH, Genentech, Gates Foundation  

**More information**

[http://hcbravo.org](http://hcbravo.org)  
[\@hcorrada](https://twitter.com/hcorrada)
    </textarea>
<script src="libs/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"ratio": "16:9"
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {window.dispatchEvent(new Event('resize'));});</script>

<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre']
  }
});
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
