---
title: "Visualization, statistical modeling and discovery in computational epigenomics"
author: |
  Héctor Corrada Bravo   
  Center for Bioinformatics and Computational Biology  
  University of Maryland, College Park  
date: Septmeber 21, 2015
output: 
  ioslides_presentation:
    css: hcb_styles.css
    widescreen: true
---

---

<div class="centered">
<img src="img/stickmen.png" style="max-height: 50%; max-width: 50%"/>
</div>

- Study the **molecular** basis of *variation* in development and disease
- Using **high-throughput** experimental methods


## The Genomic Revolution

- For over a decade we have laid the basic molecular blueprint by **sequencing** DNA

<div class="centered">
<img src="img/genome_projects.png" style="max-height: 100%; max-width: 100%"/>
</div>

## The Genomic Revolution

**NHGRI strategic plan**

<div class="centered">
<img src="img/basepair_bedside.png" style="max-height: 100%; max-width: 100%"/>
</div>

<footer class="source">[Nature, 2011]</footer>

## The Genomic Revolution

**NHGRI strategic plan**

"The major bottleneck in genome sequencing is no longer **data generation**—the computational challenges around **data analysis**, display and integration are now rate limiting. New approaches and methods are required to meet these challenges."

- Data analysis 
- Data integration
- Visualization
- Computational tools and infrastructure

<footer class="source">[Nature, 2011]</footer>

---

My group's work as a simplex

![](img/hcbravolab_simplex.png)

# Computational Epigenomics

```{r child='epigenomics.Rmd' }
```

```{r child='antiprofiles.Rmd' }
```

```{r child='methylflow.Rmd' }
```

## Moving forward

- move anti-profiles closer to the clinic
- explore _anomaly classification_ as a general learning setting
- methods to understand hierarchical organization of epigenomic domains
- better understand connection between intra-tumor heterogeneity and consistent hyper-variability in cancer

```{r child='epiviz.Rmd' }
```

## Moving forward

- collaborative computational and visual analysis (w/ N. Elmqvist @ HCIL)
- effective visual methods to explore hierarchical organization of genome
- deeper integration of statistically-informed visualization
- visualization-informed statistical analysis


---

- Discoveries: consistent hypo-methylation, hyper-variability
- Methods: anomaly classification as a setting to understand predictor stability
- Tools: computational and visual exploratory genomic data analysis

![](img/hcbravolab_simplex.png)

---

Tools

- `antiProfiles`
- `minfi`
- `bumphunter`
- `Rcplex`
- `Rcsdp`
- `BlindCall`
- `HTShape`
- `qsmooth`
- `epiviz`
- `epivizr`

## Metagenomics (mixed genomes)

<div class="centered">
<img src="img/hmp.png" style="max-width=50%; min-width=50%"/>
</div>

<footer class="source">[Human Microbiome Project]</footer>


## Metagenomics (mixed genomes)

- Discoveries: pathogenic associations for childhood diarrhea in developing world. (Genome Biology, 2014)
- Methods: association discovery for metagenomic communities. (Nature Methods, 2013)
- Tools: `metagenomeSeq`, `metagenomicFeatures`, `metaviz`

---

**Coordinates**:

<div class="centered">
<img src="img/tree.png" style="max-width=1%; min-width=1%"/>
</div>

---

**Samples**:
<div class="centered">
<img src="img/meta_heatmap.jpg" style="max-width=50%; min-width=50%"/>
</div>


---


<div class="centered">
<img src="img/data_sources_with_hierarchy.png" style="max-height: 45%; max-width: 45%"/>
</div>

Hierachically organized features

---

Hierarchically organized features

<div class="centered">
![](movies/metavizr1.gif)
</div>

---

### NHGRI strategic plan

"Meeting the computational challenges for genomics requires scientists with expertise in biology as well as in informatics, computer science, mathematics, statistics and/or engineering." 

_A new generation of investigators who are proficient in two or more of these fields must be trained and supported._

---

###Acknowledgements
Colleagues at CBCB  
Past members of HCBravo group (now at Harvard, U. Chicago, Johns Hopkins, Genentech, Dow Jones Data Science)  
Current members of HCBravo group  
Collaborators at JHU/Harvard: Jeff Leek, Rafael A. Irizarry, Andy Feinberg, Winston Timp, Kasper Hansen, and Ben Langmead 

###More information
[http://hcbravo.org](http://hcbravo.org)  
[\@hcorrada](https://twitter.com/hcorrada)    


